### ISSUE TITLE
Remove rlibc section?

### ISSUE THREAD

**Issue by cristicismas** [phil-opp/blog_os#862]
Hello! I've just started reading the blog posts, and what I've noticed is that [this section](https://os.phil-opp.com/minimal-rust-kernel/#the-rlibc-crate) might not be up to date anymore. Building with the rlibc dependency didn't work for me, so I've looked into it, and it seems like rlibc has been [deprecated](https://github.com/alexcrichton/rlibc). 

Do you think that section should be removed? I've simply skipped that section all together but I'm not sure if it will cause any problems down the road.

**Comment by 12101111** [comment]
This is blocked by https://github.com/rust-lang/wg-cargo-std-aware/issues/53

**Comment by phil-opp** [comment]
@cristicismas To my knowledge, the `rlibc` crate is still required in our case. What problems did you have with building it?

**Comment by cristicismas** [comment]
@phil-opp For some reason it didn't let me install it as a dependency. I _think_ that I was getting a 404 error, which was strange but I moved on since I didn't think `rlibc` would be needed anymore. 

I've progressed further into the blog and I've hit a wall when writing text to VGA (errors with memcpy and its friends). So when I looked into the issue I saw this on the rlibc repo:

```
The compiler-builtins crate now provides optimized versions of these functions when enabling the optional mem feature. [...] It should be used instead of this one.
```

Since we're recompiling `compiler-builtins` individually, we can't specify 'mem' as a feature in `Cargo.toml`, so we would need a way to specify it in `.cargo/config.toml`. The way to do this, as far as I know, is inside `build-std-features`, but that doesn't work yet for compiler-builtins. 

There is an open PR for it though: https://github.com/rust-lang/rust/pull/77284, so when that is merged I guess the rlibc section could be replaced with instructions on build-std-features.

**Comment by phil-opp** [comment]
Thanks for the pointer!

We used the `mem` feature of `compiler-builtins` before when we used `cargo-xbuild` for cross compiling. We moved to `rlibc` after switching to `-Zbuild-std` because, as you said, enabling the `mem` feature is not possible yet. I'm happy to switch back to it once the PR you linked is merged.

**Comment by cristicismas** [comment]
It looks like it's been merged!

**Comment by phil-opp** [comment]
Yes, I saw that! I'll try to update the blog in the next few days.

**Comment by phil-opp** [comment]
Implemented in #865, #866, and #868.
